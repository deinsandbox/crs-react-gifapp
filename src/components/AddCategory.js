import React, { useState } from "react";
import PropTypes from "prop-types";

const AddCategory = ({ setCategory }) => {
  const [text, setText] = useState("");

  const handleOnChange = (e) => {
    setText(e.target.value);
  };

  const handleSubmit = (e) => {
    e.preventDefault();
    const item = text.trim();
    if (item.length > 2) {
      setCategory((items) => [item, ...items]);
      setText("");
    }
  };

  return (
    <form onSubmit={handleSubmit}>
      <input type="text" name="" id="" value={text} onChange={handleOnChange} />
    </form>
  );
};

AddCategory.propTypes = {
  setCategory: PropTypes.func.isRequired,
};

export default AddCategory;
