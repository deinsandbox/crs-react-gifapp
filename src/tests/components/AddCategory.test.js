import React from "react";
import "@testing-library/jest-dom";
import { shallow } from "enzyme";
import AddCategory from "../../components/AddCategory";

describe("Test on <AddCategory /> component", () => {
  const value = "Hola Mundo";
  const setCategory = jest.fn();
  const setText = jest.fn();
  let wrapper;

  beforeEach(() => {
    jest.clearAllMocks();
    wrapper = shallow(<AddCategory setCategory={setCategory} />);
  });

  test("should render the component correctly", () => {
    expect(wrapper).toMatchSnapshot();
  });

  test("should change the input value", () => {
    const before = wrapper.find("input");
    before.simulate("change", { target: { value } });
    const after = wrapper.find("input");
    expect(after.prop("value")).toBe(value);
  });

  test("should not be submit category if has not valid value", () => {
    const form = wrapper.find("form");
    form.simulate("submit", { preventDefault() {} });
    expect(setCategory).not.toHaveBeenCalled();
  });

  test("should submit a category and clean the input", () => {
    const input = wrapper.find("input");
    input.simulate("change", { target: { value } });

    const form = wrapper.find("form");
    form.simulate("submit", { preventDefault() {} });

    expect(setCategory).toHaveBeenCalledTimes(1);
    expect(setCategory).toHaveBeenCalledWith(expect.any(Function));

    expect(input.prop("value")).toBeEmpty;
  });
});
