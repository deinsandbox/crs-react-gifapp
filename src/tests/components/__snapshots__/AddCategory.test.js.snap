// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test on <AddCategory /> component should render the component correctly 1`] = `
<form
  onSubmit={[Function]}
>
  <input
    id=""
    name=""
    onChange={[Function]}
    type="text"
    value=""
  />
</form>
`;
